network/gossip/libp2p/middleware.go

```diff
@@ -191,11 +191,10 @@ func (m *Middleware) Start(ov middleware.Overlay) error {
 	if err != nil {
 		return fmt.Errorf("failed to create libp2pConnector: %w", err)
 	}
+
 	m.peerManager = NewPeerManager(m.ctx, m.log, m.ov.Topology, libp2pConnector)
-	err = m.peerManager.Start()
-	if err != nil {
-		return fmt.Errorf("failed to start peer manager: %w", err)
-	}
+	<-m.peerManager.Ready()
```

Comment:
We should create a timeout for these if there isn't already a timeout further up the stack while initializing this component. IE:
```go
select {
   <-m.peerManager.Ready():
   <-time.After(time.Second*30):
         // error timeout
}
```

Reply:
Fixed in https://github.com/onflow/flow-go/pull/99/commits/14e5a0bcf79ed31b89f81d442e27954d13b256af